feat: multi Select for "Event Type" in Insights

ShashwatPS opened the pull request:
## What does this PR do?

- Fixes #23152 
- Fixes CAL-6271

## Visual Demo (For contributors especially)

### After Fix
https://www.loom.com/share/876816ba23bd4253ae38422da58efad4?sid=17dfee36-4411-4538-8818-830468165d4d

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- No environment variables needed just running it locally works.
- Creating multiple events to test the multi select.

### Files Modified  
packages/features/insights/hooks/useInsightsBookings.ts
packages/lib/server/service/InsightsBookingBaseService.ts

### Approach  
```
filter: {
            type: ColumnFilterType.MULTI_SELECT,
          },
},
```
Changed the type to MULTI_SELECT, the rest of the logic is handled internally.    

Whenever buildColumnFilterCondition is called with filter data, while handling the eventTypeId filter type, it used to check for a single number. I changed it to handle the MULTI_SELECT data type instead.

```
// File: packages/lib/server/service/InsightsBookingBaseService.ts
  if (id === "eventTypeId" && isMultiSelectFilterValue(value)) {
      const eventTypeIds = value.data.map((id) => Number(id));
      return Prisma.sql`("eventTypeId" IN (${Prisma.join(eventTypeIds)}) OR "eventParentId" IN (${Prisma.join(
        eventTypeIds
      )}))`;
}
```

## Checklist  
- [x] I have read the contributing guide  
- [x] My code follows the style guidelines of this project  
- [x] I have commented my code, particularly in hard-to-understand areas  
- [x] I have checked that my changes generate no new warnings

ShashwatPS committed:
Multi select event type in insights

anikdhabal reviewed (dismissed):
It is allowing to select multiple event-type, but i don't think it's correctly filter event-types. Pls test it

ShashwatPS committed:
Fixing multi select query generation

ShashwatPS commented:
> It is allowing to select multiple event-type, but i don't think it's correctly filter event-types. Pls test it

@anikdhabal I’ve fixed it and updated the logic, along with the video in the PR. Could you have a look whenever you get a chance ?

ShashwatPS committed:
Added null check for eventType filters

anikdhabal commented:
@ShashwatPS test is failing

ShashwatPS commented:
@anikdhabal The tests needs to changed for this, should I go ahead and make the change ?

ShashwatPS committed:
Updated test to support multiselect

ShashwatPS committed:
Type fix

Devanshusharma2005 committed:
Merge branch 'main' into shashwatps/multi_select_event

eunjae-lee reviewed (approved):
Working well. Thanks for the work!

eunjae-lee committed:
Merge branch 'main' into shashwatps/multi_select_event
